####-rfc-docs

```diff
@@ -0,0 +1,126 @@
+
+### Phase 2 - Update qiskit.org
+
+The following features will need to be added to qiskit.org:
+
+- a mechanism for fetching the json data for docs pages
+- ability to dynamically render multiple docs pages, one for each json file fetched
+- a standard docs page structure, including restyling any sphinx html objects for the main body of the page
+
+### Phase 3 - Update Qiskit main package repos
+
+The main package repos will need to be updated to: 
+
+- remove the `qiskit_sphinx_theme` as a dependency
+- build their documentation files using the sphinx json builder insteda of the html builder
+- (maybe?) change the docs deployment process to deploy the json files somewhere (this step still TBD)
+
+This phase would start as a pilot with one Qiskit repo before having others follow the same implementation
+
+
+## Detailed Design
+
+1. To generate the sphinx json files I ran `sphinx-build -b json docs/ docs/_build_json/json`
+    - the files produced by the sphinx build include the following:
+        - a `globalcontext.fjson` which allows us to access any context variables set by individual Qiskit repos in their `docs/conf.py`
+        - a `searchindex.fjson` which should help us deal with search somehow (but not sure how yet)
+        - one `.fjson` file for each `.rst` file, which includes some TOC data and a `body` attribute consisting of an html string containing the main body of the page.
+2. For investigation purposes I just manually copied over the files into my fork of the qiskit.org repo (I had to change the file names from `.fjson` to `.json`)
+3. You can see my first pass at integrating the data into qiskit.org in my draft PR [here](https://github.com/Qiskit/qiskit.org/pull/2858).
+
+
+Here is a first screenshot of some experimenting with integrating a docs page into qiskit.org:
+
+<img width="1785" alt="Screenshot 2022-10-25 at 5 58 03 PM" src="https://user-images.githubusercontent.com/23662430/198141449-4a2298a8-9796-4105-a25b-9fa20ece4a0d.png">
+
+### Challenges/Limitations so far
+
+some initial challenges/drawbacks have been identified as part of the investigation phase so far:
+- each `json` file has a `body` attribute which contains the main content of the page (i.e. what gets rendered in the middle section of the screenshot). This attribute is a long html string that we can render with Vue BUT we need to go manually change all the styling. Doing this for every possible html tag and class name used in all the docstrings will be a large piece of work. The screenshot is pretty similar to the raw output, I haven't adjusted much of the css there.
+- Internal links within the `body` object get added to the html like this: `href="../../stubs/qiskit.circuit.QuantumCircuit/#qiskit.circuit.QuantumCircuit" `, which is not ideal, we’ll have to figure out how to adjust that for every link on every page
+- The sphinx build output is actually `.fjson` which is exactly like `.json` but with an f in front of it. This could maybhe cause issues?
```

Comment:
FWIW,The built in json builder (along with the other built-in serialized builder using pickle) is basically designed to host the content from a custom web app instead of being consumed as static content on a web server directly. That's why it has the html body embedded as an object field. You might be better off just writing a custom build to output content the way you want if this is the route you're pursuing.